.github/workflows/pinot_vuln_check.yml

```diff
@@ -0,0 +1,66 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: Pinot Dependencies
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+    paths-ignore:
+      - "contrib/**"
+      - "docs/**"
+      - "licenses/**"
+      - "licenses-binary/**"
+      - "**.md"
+jobs:
+  verify-docker:
+    name: Verify Docker Image
+    runs-on: ubuntu-latest
+    steps:
+      - uses: docker/setup-qemu-action@v1
+        name: Set up QEMU
+      - uses: docker/setup-buildx-action@v1
+        name: Set up Docker Buildx
+      - uses: actions/checkout@v3
+      - name: Build the Docker image
+        env:
+          DOCKER_FILE_BASE_DIR: "docker/images/pinot"
+          DOCKER_IMAGE_NAME: "apachepinot/pinot"
+          BUILD_PLATFORM: "linux/amd64"
+          PINOT_GIT_URL: ${{ github.event.inputs.gitUrl }}
+          PINOT_BRANCH: ${{ env.GITHUB_REF }}
+          TAGS: ${{ github.sha }}
+        run: .github/workflows/scripts/docker/.pinot_docker_image_build.sh
+
+      - name: Run Trivy vulnerability scanner (sarif)
+        uses: aquasecurity/trivy-action@master
+        with:
+          trivyignores: '.trivyignore'
+          image-ref: 'apachepinot/pinot:${{ github.sha }}'
+          format: 'sarif'
+          output: 'trivy-results.sarif'
+          vuln-type: 'os,library'
+          severity: 'CRITICAL,HIGH'
+      - name: Upload Trivy scan results to GitHub Security tab
+        uses: github/codeql-action/upload-sarif@v2
```

Comment:
is there an option to not create a failed GHA task but instead comment on the PR like the codecov bot?

Reply:
I didn't find it. But the task compares the vulns in source and target branch. If no new vulnerability is added, it should be empty even if we have vulns in master. At least that is what I infer from the code scanning result.

I would recommend to merge the PR as it is to make it clear there are some vulnerabilities. We can add to .trivyignore the once we don't plan to fix soon.